concent_api/core/payments/bankster.py

```diff
@@ -383,24 +390,82 @@ def sum_subtask_price(subtask_results_accepted_list: List[SubtaskResultsAccepted
 
 
 def get_provider_payment_info(
-    list_of_forced_payments: List[ForcedPaymentEvent],
-    list_of_payments: List[BatchTransferEvent],
+    list_of_settlement_payments: List[ForcedPaymentEvent],
+    list_of_transactions: List[BatchTransferEvent],
+    settlement_payment_claims: QuerySet,
     subtask_results_accepted_list: List[SubtaskResultsAccepted],
 ) -> Tuple[int, int]:
-    assert isinstance(list_of_payments, list)
-    assert isinstance(list_of_forced_payments, list)
+    assert isinstance(settlement_payment_claims, QuerySet)
+    assert isinstance(list_of_settlement_payments, list)
+    assert isinstance(list_of_transactions, list)
     assert isinstance(subtask_results_accepted_list, list)
 
-    force_payments_price = sum_payments(list_of_forced_payments)
-    payments_price = sum_payments(list_of_payments)
+    settlement_payment_total_price = sum_payments(list_of_settlement_payments)
+    payments_price = sum_payments(list_of_transactions)
+    satisfied_payments_price = settlement_payment_claims.aggregate(
+        sum_of_already_satisfied_claims=Coalesce(Sum('amount'), 0)
+    )['sum_of_already_satisfied_claims']
     subtasks_price = sum_subtask_price(subtask_results_accepted_list)
 
-    amount_paid = payments_price + force_payments_price
+    amount_paid = payments_price + satisfied_payments_price + settlement_payment_total_price
     amount_pending = subtasks_price - amount_paid
 
     return (amount_paid, amount_pending)
 
 
+def remove_duplicated_transactions_from_deposit_claims(
+    already_satisfied_claims: QuerySet,
+    list_of_settlement_payments: List[ForcedPaymentEvent],
+) -> QuerySet:
+    """
+    Removes duplicates from claims of settlement payments with pending transaction
+    which are on list of forced payments by adding exclusions to DepositClaim queryset.
+    """
+    settlement_payment_tx_hashes = [force_payment.tx_hash for force_payment in list_of_settlement_payments]
+
+    for matching_claim in already_satisfied_claims.filter(tx_hash__in=settlement_payment_tx_hashes):
+        assert matching_claim.concent_use_case == ConcentUseCase.FORCED_PAYMENT
+
+        matching_force_payment = [
+            force_payment for force_payment in list_of_settlement_payments
+            if force_payment.tx_hash == matching_claim.tx_hash
+        ][0]
```

Comment:
You should use `next()` here. You give it an iterator and it either returns the first element or raises `StopIteration` (if there are no elements). It's a more readable and also a tiny bit more efficient because it does not need to construct the whole list just to discard everything but the first element.

Reply:
:+1: done